bravado/http_future.py

```diff
@@ -136,9 +135,25 @@ def response(self, timeout=None, fallback_result=None, exceptions_to_catch=FALLB
         try:
             incoming_response = self._get_incoming_response(timeout)
             request_end_time = monotonic.monotonic()
+
+            # Trigger fallback_result if the option is set
+            if fallback_result and self.request_config.force_fallback_result:
+                if self.operation.swagger_spec.config['bravado'].disable_fallback_results:
+                    log.warning(
+                        'force_fallback_result set in request options and disable_fallback_results '
+                        'set in client config; not using fallback result.'
+                    )
+                else:
+                    # raise an exception to trigger fallback result handling
+                    fake_exception_type = exceptions_to_catch[0]
+                    args = (incoming_response,) if issubclass(fake_exception_type, HTTPError) else ()
+                    raise fake_exception_type(*args)
```

Comment:
I'm wondering it this will always be possible and I think that it is not.

At the moment you're customising the value of `args` in case of an `HTTPError`; what happen if the first exception in `exceptions_to_catch` needs additional arguments to be instantiated?
An example could be [`aiohttp.client_exceptions.ClientResponseError`](https://github.com/aio-libs/aiohttp/blob/965ddc1ea3a8fbf1f8d0f64efa0f117a8ebb64ab/aiohttp/client_exceptions.py#L43)

Other than potential issues in creating the exception to raise this could have negative effects over the monitoring systems that we could set as we'll not be able to distinguish if the exception was a "real" exception or an injected one.

To workaroud those issues I would suggest to define a custom exception (ie. `DegradedResponseError`), raise it and ensure that only that exception will be catched.
This allows you to:
 * not have to care about the initialisation of the exception instance (as it will be under complete bravado control)
 * be able to discriminate on the logs what exception have been raised (so monitoring systems will not alert the teams in case of a flood of requests which forces force_degradation)

Reply:
@macisamuele good points, and as we've discussed I'm not sure about what the best approach overall is. Earlier on I used a BravadoTimeoutError; the problem with not using an exception from `exceptions_to_catch` is that the `fallback_result` function is not guaranteed to be able to handle that type of exception.

This PR gives users some control over which exception is going to be thrown in the `force` case, although I'll admit it's quite unintuitive and doesn't cover all cases.

If we're ok with introducing this new custom exception type then I'll go ahead and change this PR accordingly - users can then adapt their callback functions to treat it any way they want to.